src/main/java/SuperPeer.java

```diff
@@ -1,55 +1,98 @@
 /*
  * This Java source file was generated by the Gradle 'init' task.
  */
+import ether.TransactionsManager;
 import io.left.rightmesh.id.MeshID;
 import io.left.rightmesh.mesh.JavaMeshManager;
 import io.left.rightmesh.mesh.MeshManager;
-import io.left.rightmesh.mesh.MeshManager.MeshTransactionEvent;
-import io.left.rightmesh.util.MeshUtility;
+import io.left.rightmesh.mesh.MeshStateListener;
+import io.left.rightmesh.util.RightMeshException;
+import io.reactivex.functions.Consumer;
 
-import java.io.Console;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
 
+import static io.left.rightmesh.mesh.MeshManager.PEER_CHANGED;
 import static io.left.rightmesh.mesh.MeshManager.TRANSACTION_RECEIVED;
- 
-public class SuperPeer {
 
+public class SuperPeer implements MeshStateListener {
+
+    //TODO: Add logger instead of system.out
     public static final String TAG = SuperPeer.class.getCanonicalName();
-        
+
+    private boolean isRunning = true;
+    private TransactionsManager tm;
+
     public static void main(String[] args) {
         SuperPeer p = new SuperPeer();
     }
-    
+
     public SuperPeer() {
         JavaMeshManager mm = new JavaMeshManager(true);
-        mm.on(TRANSACTION_RECEIVED, this::handleTransactionPacket);
         System.out.println("Superpeer MeshID: " + mm.getUuid());
-        
+
         System.out.println("Waiting for lib to be ready");
         try {
-                Thread.sleep(200);
-        } catch(Exception ex) {}
-        
-        //test send on the TRANSACTION PORT
-        mm.sendDataReliable(mm.getUuid(), MeshUtility.TRANSACTION_PORT, "test".getBytes());
-        
-        //Continue to run until user quits
-        System.out.println("Enter empty line to quit");
-        Console c = System.console();
-        String msg;
-        if (c == null) {
-            System.err.println("No console.");
-            mm.stop();
-            return;
+            Thread.sleep(200);
+        } catch (Exception ex) {
```

Comment:
Catching Exception is a bad habit. Checkstyle might yell at you for it too.

Reply:
done